packages/flutter/lib/src/cupertino/page.dart

```diff
@@ -26,6 +25,20 @@ final FractionalOffsetTween _kBottomUpTween = new FractionalOffsetTween(
   end: FractionalOffset.topLeft,
 );
 
+// BoxDecoration from no shadow to page shadow mimicking iOS page transitions.
+final DecorationTween _kShadowTween = new DecorationTween(
+  begin: BoxDecoration.noDecoration, // No shadow initially.
```

Comment:
DecorationTween doesn't override evaluate and Tween just return the begin/end as is at the edges. For DecorationBoxes, the decoration is expected to be non-null.

There was a couple of ways. Removing the non-null requirement on DecorationBox is a bit too invasive. We can make DecorationTween always lerp from the non-null value but it's already used in a bunch of places expecting the actual begin/end null values in AnimatedWidgetBaseState. This way felt the cleanest.